Core/Base/API.py

```diff
@@ -65,21 +65,31 @@ def __init__( self ):
 
   #############################################################################
 
-  def __getstate__( self ):
-    """ called when pickling the object. Also used in copy.deepcopy """
-    from DIRAC.FrameworkSystem.private.logging.SubSystemLogger import SubSystemLogger
-    d = dict( self.__dict__ )
-    if isinstance( d['log'], SubSystemLogger ):
-      d['log'] = d['log']._subName
-    return d
+  def __getstate__(self):
+    """ Return a copied dictionary containing all the attributes of the API.
+        Called when pickling the object. Also used in copy.deepcopy.
 
-  #############################################################################
-
-  def __setstate__( self, d ):
-    """ called when un-pickling the object """
-    self.__dict__.update(d)
-    if isinstance( d['log'], basestring ):
-      self.log = gLogger.getSubLogger( d['log'] )
+    :return: dictionary of attributes
+    """
+    from DIRAC.FrameworkSystem.private.standardLogging.Logging import Logging
+    state = dict(self.__dict__)
+    # Replace the Logging instance by its name because it is not copyable
+    # because of the thread locks
+    if isinstance(state['log'], Logging):
```

Comment:
To keep this backward compatible, also check for SubSystemLogger instance (keeping the other import)
```
if isinstance(state['log'], (SubSystemLogger, Logging) ):
```